context/app/static/js/components/DatasetDetail.jsx

```diff
@@ -61,45 +48,46 @@ function DatasetDetail(props) {
     assayMetadata,
     provData,
     vitData,
-    flashed_messages,
     assetsEndpoint,
+    flashed_messages,
   } = props;
   const {
-    protocol_url, portal_uploaded_protocol_files, metadata, files, uuid, data_types, origin_sample,
+    protocol_url,
+    portal_uploaded_protocol_files,
+    metadata,
+    files,
+    uuid,
+    data_types,
+    origin_sample,
   } = assayMetadata;
+
+  const displayViz = 'name' in vitData;
+  const displayProtocol = (portal_uploaded_protocol_files || protocol_url);
+  const displayMetadataTable = (metadata && metadata.metadata);
+  const displayFiles = (files && files.length);
+  const displayDag = (metadata && (metadata.dag_provenance || metadata.dag_provenance_list));
```

Comment:
Preference for boolean variables that sound like questions... Often `is...`, but in this case maybe `should...` would work.